draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -2712,28 +2710,150 @@ of the counter byte I2OSP(i, 1)).
 
 The essential difference between the construction of {{CDMP05}} and
 expand\_message\_xmd is that the latter hashes a counter appended to
-strxor(b\_0, b\_(i - 1)) (step 9) rather than to b\_0.
+strxor(b\_0, b\_(i - 1)) (step 10) rather than to b\_0.
 This approach increases the Hamming distance between inputs to different
 invocations of H, which reduces the likelihood that nonidealities in H
 affect the distribution of the b\_i values.
 
+We note that expand\_message\_xmd can be used to instantiate a general-purpose
+indifferentiable functionality with variable-length output based on any hash
+function meeting one of the above criteria.
+Applications that use expand\_message\_xmd outside of hash\_to\_field should
+ensure domain separation by picking a distinct value for DST.
+
+## Domain separation recommendations {#security-considerations-domain-separation}
+
+The expand\_message variants in this document ({{hashtofield-expand}})
+always append a suffix-free-encoded domain separation tag DST\_prime
+to the strings hashed by H, the underlying hash or extensible output function.
+(Other expand\_message variants that follow the guidelines in
+{{hashtofield-expand-other}} are expected to behave similarly,
+but these should be analyzed on a case-by-case basis.)
+Protocols that also use H outside of expand\_message should enforce domain
+separation between those uses of H and expand\_message, and should separate
+all of these from uses of H outside the protocol.
+The purpose of enforcing domain separation is to ensure that security analyses
```

Comment:
I think this line could be better sited at the start of the section. 
So, first recalling the objective of domain separation, 
then showing how we achieve this objective _inside_ expand_message,
and then pointing out how others can  _externally_ use H preserving domain separation.